vnc_networks/connections.py

```diff
@@ -52,46 +57,89 @@
     "output_clustering",
 ]
 
-## ---- Constants ---- ##
-FIGSIZE = (8, 8)
-DPI = 300
-
 ## ---- Classes ---- ##
+class Connections:
+    def __deepcopy__(self, memo):
+        """
+        deepcopy everything except the ConnectomeReader instance that can
+        be shared between the original and the copy
+        """
+        new_instance = self.__class__.__new__(self.__class__)
+        memo[id(self)] = new_instance
 
+        for k, v in self.__dict__.items():
+            if k == "CR":
+                setattr(new_instance, k, v)
+            else:
+                setattr(new_instance, k, copy.deepcopy(v, memo))
+        
+        return new_instance
 
-class Connections:
     def __init__(
         self,
+        CR: Optional[ConnectomeReader] = ConnectomeReader('MANC','v1.0'),
+        from_file: Optional[str] = None,
         neurons_pre: Optional[list[int] | list[BodyId]] = None,
         neurons_post: Optional[list[int] | list[BodyId]] = None,
-        from_file: Optional[str] = None,
+        nt_weights: Optional[Mapping[str, int]] = params.NT_WEIGHTS,
+        split_neurons: Optional[list[Neuron]] = None,
+        not_connected: Optional[list[BodyId] | list[int]] = None,
+        keep_only_traced_neurons: Optional[bool] = True,
```

Comment:
Again, I don't think `nt_weights` and `keep_only_traced_neurons` should be `Optional` here if there are valid default values. I left a longer comment at the top of `neuron.py` about this